R/caretList.R

```diff
@@ -319,55 +318,91 @@ as.caretList.list <- function(object){
 #' @importFrom pbapply pboptions
 #' @export
 #' @method predict caretList
-predict.caretList <- function(object, newdata = NULL, ..., verbose = FALSE){
-
-  if(is.null(newdata)){
+predict.caretList <- function(object, newdata = NULL, ..., verbose = FALSE) {
+  if (is.null(newdata)) {
     warning("Predicting without new data is not well supported.  Attempting to predict on the training data.")
     newdata <- object[[1]]$trainingData
-    if(is.null(newdata)){
+    if (is.null(newdata)) {
       stop("Could not find training data in the first model in the ensemble.")
     }
   }
 
-  if(verbose == TRUE){
+  if (verbose == TRUE) {
     pboptions(type = "txt", char = "*")
-  } else if(verbose == FALSE){
+  } else if (verbose == FALSE) {
     pboptions(type = "none")
   }
-  preds <- pbsapply(object, function(x){
+  preds <- pbsapply(object, function(x) {
     type <- x$modelType
-    if (type=="Classification"){
-      if(x$control$classProbs){
-        # Return probability predictions for only one of the classes
-        # as determined by configured default response class level
-        caret::predict.train(x, type="prob", newdata=newdata, ...)[, getBinaryTargetLevel()]
-      } else{
-        caret::predict.train(x, type="raw", newdata=newdata, ...)
+    if (type == "Classification") {
+      if (x$control$classProbs) {
+        caret::predict.train(x, type = "prob", newdata = newdata, ...)
+      } else {
+        caret::predict.train(x, type = "raw", newdata = newdata, ...)
       }
-    } else if(type=="Regression"){
-      caret::predict.train(x, type="raw", newdata=newdata, ...)
-    } else{
+    } else if (type == "Regression") {
+      caret::predict.train(x, type = "raw", newdata = newdata, ...)
+    } else {
       stop(paste("Unknown model type:", type))
     }
   })
-  if(class(preds) != "matrix" & class(preds) != "data.frame"){
-    if(class(preds) == "character" | class(preds) == "factor"){
+
+  if (!inherits(preds, "matrix") && !inherits(preds, "data.frame")) {
+    if (inherits(preds, "character") || inherits(preds, "factor")) {
       preds <- as.character(preds) # drop factorization
     }
     preds <- as.matrix(t(preds))
   }
 
-  if (is.null(names(object))){
+  if (is.null(names(object))) {
     # If the model list used for predictions is not currently named,
     # then exctract the model names from each model individually.
     # Note that this should only be possible when caretList objects
     # are created manually
-    predcols <- sapply(object, extractModelName)
-    colnames(preds) <- make.names(predcols, unique=TRUE)
+    modelnames <- make.names(sapply(object, extractModelName), unique = TRUE)
+  } else {
+    # Otherwise, assign the names of the prediction columns
+    # using the names in the given model list
+    modelnames <- names(object)
+  }
+
+  if (object[[1]]$modelType == "Classification" && object[[1]]$control$classProbs) {
```

Comment:
This is covered in the new tests, correct?

Reply:
Yes